TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opMode/teleOp/BlueDriverOpMode.java

```diff
@@ -2,12 +2,16 @@
 
 import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
 
+import org.firstinspires.ftc.teamcode.TeleOpRobotController;
 import org.firstinspires.ftc.teamcode.util.TeamColor;
+import org.firstinspires.ftc.teamcode.util.opModes.SympleCommandOpMode;
 
 @TeleOp(name = "Blue Team - Driver")
-public class BlueDriverOpMode extends DriverOpMode {
+public class BlueDriverOpMode extends SympleCommandOpMode {
     @Override
     public void initialize() {
-        super.initialize(TeamColor.BLUE);
+        this.robotController = new TeleOpRobotController.Builder(this)
```

Comment:
Initialize basic properties of opmode inside a method instead of the contractor.

READBILITY!!! OR MAKOT <3

Reply:
fixed but in a weird way, please check and review the new [changes](https://github.com/SympleOrg/SympleRobotTemplate/pull/1/commits/1db74d5816d3ae4e1c8804efed7d931fed2b6b9e)

Reply:
The use of the `initializeDefaults` function is indeed more declarative but with a bit more boilerplate.
Personally I think its fine.

Reply:
The methods from the base builder is returning the type **RobotControllerBase.Builder**,
For example if you use **TeleOpRobotController.Builder#hardwareMap**, it return the type **RobotControllerBase.Builder** instead of **TeleOpRobotController.Builder** and you cannot use any method from **TeleOpRobotController.Builder** after that

Reply:
Cant we upcast to TeleOpRobotController if they are connected by inheritence?

Reply:
We can, but it will look ugly

Reply:
So lets talk about it in the meeting.

Maybe we can add another parameter to the function that tells the the object what it should be.